libenv/unix_iface.c

```diff
@@ -677,6 +675,45 @@ static void FindV6InterfacesInfo(EvalContext *ctx, Rlist **interfaces, Rlist **h
             }
         }
 
+        if (strcasestr(ifconfig_line, "ether"))
+        {
+            char *hw_mac = strtok(ifconfig_line, " ");
+            if (hw_mac == NULL) {
+              Log(LOG_LEVEL_ERR, "Failed to parse hw_mac address");
+            }
+            hw_mac = strtok(NULL, " ");
+            if (hw_mac == NULL) {
+              Log(LOG_LEVEL_ERR, "Failed to parse hw_mac address");
+            }
```

Comment:
[`strtok` is not particularly readable or safe](http://man7.org/linux/man-pages/man3/strtok.3.html#BUGS). I'd prefer something like `SeqStringFromString()`:

```C
Seq *const ether_line = SeqStringFromString(ifconfig_line, " ");
const size_t length = SeqLength(ether_line);

if (length < 2 || !StringSafeEqualIgnoreCase(SeqAt(ether_line, 0), "ether"))
{
    Log(LOG_LEVEL_ERR, "Failed to parse hw_mac address");
}

const char *const hw_mac = SeqAt(ether_line, 1);

if (!RlistContainsString(*hardware, hw_mac))
{
    [...]
}
SeqDestroy(ether_line);
```

Reply:
I knew there was a comment inbound for this one :p